src/plots/cartesian/set_convert.js

```diff
@@ -152,10 +152,8 @@ module.exports = function setConvert(ax, fullLayout) {
         if(ax._categoriesMap) {
             index = ax._categoriesMap[v];
             if(index !== undefined) return index;
-        } else {
-            index = ax._categories.indexOf(v);
-            if(index !== -1) return index;
```

Comment:
Oh I get it. Sorry I misunderstood your point.

Well, I feel a liiiittle bit uncomfortable of doing that, because if there is any case that the map is not created, it will result in an uncaught exception, which to me, seems more severe than a messed up chart.

And I tried on [jsperf](https://jsperf.com/check-null-overhead/1), the overhead of checking null is less than 10% on Chrome:
![image](https://cloud.githubusercontent.com/assets/3392907/24883737/3f929f8c-1e14-11e7-922e-f8719605097a.png)
And considering the performance of reading a map itself, 10% differences here will not be a big cost for the whole workflow.

For Safari the differences are even less.
![image](https://cloud.githubusercontent.com/assets/3392907/24883693/0f89de40-1e14-11e7-8aec-69fd77275a84.png)